### PR TITLE
Introducing sem ver

### PR DESCRIPTION
**Pull request by aryszka**
DON'T MERGE JUST YET

Before merging, we have to tag an initial version. I'd like to start with v0.0.9, is it ok?

How it works:

1. on pushing to branches, it runs the tests as currently
2. on creating a pull request, it runs the tests, checks if formatting was applied, and runs go vet
3. merge:
a) when merging to master, it bumps the patch version
b) when merging to master, and the commit message contains "minor-release" (the branch name is automatically picked when merging), bumps the minor version
c) like 3b but for the major version

### CODE REVIEW

**Comment by lmineiro** [comment]
NICE!

File: Makefile [hunk 1]
```diff
@@ -17,15 +35,58 @@ shortcheck: build
...
 	go vet ./...
 
-fmt:
+fmt: $(SOURCES)
 	gofmt -w $(SOURCES)
 
-checkfmt:
-	if [[ $$(gofmt -d $(SOURCES)) ]]; then false; else true; fi
+check-fmt: $(SOURCES)
+	if [ "$$(gofmt -d $(SOURCES))" != "" ]; then false; else true; fi
```
Comment (mo-gr) [inline Makefile]: couldn't this be achieved with the `test` command? An if/else that does nothing but return true/false always looks weird to me

File: packaging/version/version.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	case cmdPatch:
+		*patch++
+	}
+}
+
+func main() {
+	cmd, current, err := command()
+	if err != nil {
+		log.Println(usage())
+		return
```
Comment (mo-gr) [inline packaging/version/version.go]: return with non-zero return code maybe?

**Comment by mo-gr** [comment]
Awesome!

File: packaging/version/version.go (hunk 2 above)
Comment (aryszka) [inline packaging/version/version.go]: right, os even imported already

Comment (aryszka) [inline packaging/version/version.go]: log.Fatalln even better

**Pull request merged**